mobile/src/bundles/common/components/autocomplete-multi-selector/autocomplete-multi-selector.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+import { TextInput } from 'react-native';
+
+import {
+    ScrollView,
+    Tag,
+    Text,
+    TouchableOpacity,
+    View,
+} from '~/bundles/common/components/components';
+import { IconName, TextCategory } from '~/bundles/common/enums/enums';
+import {
+    useCallback,
+    useFormController,
+    useMemo,
+    useState,
+} from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/styles';
+
+import { styles } from './styles';
+
+type Properties<T extends FieldValues> = {
+    control?: Control<T, null>;
+    name: FieldPath<T>;
+    hasError?: boolean;
+    items: string[];
+    placeholder?: string;
+};
+
+const AutocompleteMultiSelector = <T extends FieldValues>({
+    control,
+    name,
+    hasError,
+    items,
+    placeholder,
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+    const { value, onBlur } = field;
+    const [isListVisible, setIsListVisible] = useState<boolean>(false);
+
+    const toggleIsListVisible = (): void => {
+        setIsListVisible((previous) => !previous);
+    };
```

Comment:
I saw this logic in different places, you can create useVisibility hooks to manage visibility state ?